iothub/device/tests/PlugAndPlay/PnpHelperTests.cs

```diff
@@ -0,0 +1,161 @@
+        {
+            // Format:
+            //  {
+            //      "samplePropertyName": {
+            //          "value": 20,
+            //          "ac": 200,
+            //          "av": 5,
+            //          "ad": "The update was successful."
+            //      }
+            //  }
+
+            string propertyName = "someName";
+            int propertyValue = 10;
+            int ackCode = 200;
+            long ackVersion = 2;
+
+            string patch = PnpHelper.CreatePropertyEmbeddedValuePatch(propertyName, JsonConvert.SerializeObject(propertyValue), ackCode, ackVersion);
+
+            var jObject = JObject.Parse(patch);
+            EmbeddedPropertyPatch actualPatch = jObject.ToObject<EmbeddedPropertyPatch>();
+
+            // The property patch object should have "value", "ac", "ad" properties set. Since we did not supply an "ackDescription", "ad" should be null.
+            actualPatch.Value.SerializedValue.Should().Be(JsonConvert.SerializeObject(propertyValue));
+            actualPatch.Value.AckCode.Should().Be(ackCode);
+            actualPatch.Value.AckVersion.Should().Be(ackVersion);
+            actualPatch.Value.AckDescription.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void CreateComponentPropertyEmbeddedValuePatch()
+        {
+            // Format:
+            //  {
+            //      "sampleComponentName": {
+            //          "__t": "c",
+            //          "samplePropertyName": {
+            //              "value": 20,
+            //              "ac": 200,
+            //              "av": 5,
+            //              "ad": "The update was successful."
+            //          }
+            //      }
+            //  }
+
+            string componentName = "someComponentName";
+            string propertyName = "someName";
+            int propertyValue = 10;
+            int ackCode = 200;
+            long ackVersion = 2;
+
+            string patch = PnpHelper.CreatePropertyEmbeddedValuePatch(
+                propertyName,
+                JsonConvert.SerializeObject(propertyValue),
+                ackCode,
+                ackVersion,
+                componentName: componentName);
+
+            var jObject = JObject.Parse(patch);
+            JObject component = jObject.Value<JObject>(componentName);
+
+            // There should be two properties added to the component- the above property and a component identifier "__t": "c".
+            component.Count.Should().Be(2);
+            component[PnpHelper.PropertyComponentIdentifierKey].Should().NotBeNull();
+            ((string)component[PnpHelper.PropertyComponentIdentifierKey]).Should().Be(PnpHelper.PropertyComponentIdentifierValue);
+
+            // The property patch object should have "value", "ac", "ad" properties set. Since we did not supply an "ackDescription", "ad" should be null.
+            EmbeddedPropertyPatch actualPatch = component.ToObject<EmbeddedPropertyPatch>();
```

Comment:
What the helper really returns is a string. So we should just test for the exact string returned instead of doing a deserialization to the class defined below.

Reply:
I wouldn't rely on json string matching to do validations. There could be extra spaces, ordering differences etc; which will give rise to false negatives. 
It is always better to deserialize it into a json type, and then do validations.